docs/contributing/Building, Debugging, and Testing on Windows.md

```diff
@@ -0,0 +1,91 @@
+# Required Software
+
+1. [Visual Studio 2015 with Update 1](http://go.microsoft.com/fwlink/?LinkId=691129). _You need Update 1_.
+2. Visual Studio 2015 Extensibility Tools. If you already installed Visual Studio, choose "Modify" from the Programs and Features control panel, and check "Visual Studio Extensibility".
+
+# Getting the Code
+
+1. Clone https://github.com/dotnet/roslyn
+2. Run the "Developer Command Prompt for VS2015" from your start menu.
+3. Run `Restore.cmd` in the command prompt to restore NuGet packages.
+4. Due to [Issue #5876](https://github.com/dotnet/roslyn/issues/5876), you should build on the command line before opening in Visual Studio. Run `msbuild /v:m /m Roslyn.sln`
```

Comment:
It's fine building Roslyn using the toolset compilers locally, as long as one does not make changes to the compilers, and/or does not use any new language feature introduced by the compiler changes. This will also lose any compiler bug fixes or optimizations since.

But if one is changing compiler, it's better use Roslyn itself as the first real world project to test it out, and it's not surprising. I'm not sure what you worried about.

It's fine keeping things how they are, I was just suggesting document the different procedure when one intends to change the compilers.